x/merkledb/key.go

```diff
@@ -113,9 +113,9 @@ func (k Key) HasPrefix(prefix Key) bool {
 
 	// check that the tokens in the partially filled final byte of [prefix] are
 	// equal to the tokens in the final byte of [k].
-	remainderBitsMask := byte(0xFF << (8 - remainderTokensCount*int(k.tokenBitSize)))
-	prefixRemainderTokens := prefix.value[len(prefix.value)-1] & remainderBitsMask
-	remainderTokens := k.value[len(prefix.value)-1] & remainderBitsMask
+	remainderBitsMask := byte(0xFF >> remainderTokensCount)
```

Comment:
We used to `AND` the right bits with 0s to make the right bits 0s, but now we `OR` with 1s to make the right bits 1s. We're just checking for equality so doesn't matter either way.